src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
 
+type Shape = 'triangle'| 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
 
-}
+    const sides = [a, b, c];
+
+    sides.sort((l, k) => l - k);
+
+    const longestSide = sides.pop();
 
-export class Circle {
+    if (longestSide! >= sides[0] + sides[1]) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+  }
 
```

Comment:
you could unit these two verifications of error cases in one place
if (a<=0 || .....
a + b <= c ||.....